api/src/controllers/team.js

```diff
@@ -310,6 +315,102 @@ async function updateTeam(req, res, next) {
   }
 }
 
+/**
+  * Clear the team Data from the database on /team/:team_id
+  * @param {} req request object, containing team id in the url
+  * @param {*} res response object, the deleted team document
+  * @returns 200: teamn is deleted
+  * @returns 404: team is not found
+  * @returns 400: team id is not in a valid hexadecimal format
+  */
+async function clearTeam(req, res, next) {
+  try {
+    const { team_id: _id } = req.params;
+    const { isDeleteFlag } = req.body;
+    await HomePage.deleteMany({ teamId: _id });
+    await Website.deleteMany({ teamId: _id });
+    await Achievement.deleteMany({ teamId: _id });
+    await Publication.deleteMany({ teamId: _id });
+    if (isDeleteFlag === true) {
+      await Team.findByIdAndDelete(_id);
+      res.status(200).json('Deleted successfully!');
+    } else res.status(200).json('Cleared successfully!');
+  } catch (error) {
+    return next(
+      fillErrorObject(500, 'Error occurred with server', [error.message]),
+    );
+  }
+}
+
+/**
+  * Delete the team from the database on /team/:team_id
+  * @param {} req request object, containing team id in the url
+  * @param {*} res response object, the deleted team document
+  * @returns 200: team is deleted
+  * @returns 404: team is not found
+  * @returns 400: team id is not in a valid hexadecimal format
+  */
+async function deleteGHPages(req, res, next) {
+  const { ghToken } = req.body;
+  // Call github API to get username
+  const { data } = await axios.get('https://api.github.com/user',
+    {
+      headers: { Authorization: `token ${ghToken}` },
+    });
+  if (data.errors) {
+    return next(
+      fillErrorObject(400, 'Validation error: Repo doesnt exist!', [data.errors[0].detail]),
+    );
+  }
+
+  // Creating repoName
+  const ghUsername = data.login;
+  logger.info(`GitHub Pages delete initiated for user: ${ghUsername}`);
+  const repoName = `${ghUsername}.github.io`;
+  let validFlag = false;
+  try {
+    const repoValidator = await axios.get(`https://api.github.com/repos/${ghUsername}/${repoName}`, {
+      headers: {
+        Authorization: `token ${ghToken}`,
+        Accept: 'application/vnd.github.v3+json',
+      },
+    });
+    if (repoValidator.status === 200) validFlag = true;
+    else validFlag = false;
+  } catch (error) {
+    validFlag = false;
+  }
```

Comment:
I suggest making this a middleware so that this function's logic doesn't get too long

Reply:
Because if it doesn't exist, it shouldn't even try to delete it right? So it can return early